src/app/server/AppDelegate.h

```diff
@@ -39,4 +41,13 @@ class AppDelegate
      */
     virtual void OnCommissioningWindowOpened() {}
     virtual void OnCommissioningWindowClosed() {}
+    /*
+     * The below methods related to CASE and PASE are for the responder side of relevant communications.
+     * The methods expose the notifications related to CASE and PASE sessions on the application side.
+     */
+    virtual void OnCommissioningSessionEstablishmentStarted() {}
+    virtual void OnCommissioningSessionEstablishmentFailed(CHIP_ERROR err) {}
+    virtual void OnCASEStarted() {}
+    virtual void OnCASEFailed(CHIP_ERROR err) {}
+    virtual void OnCASEComplete(uint32_t val1, uint32_t val2) {}
```

Comment:
A few questions:

1. How will this handle a future world in which we support multiple parallel CASE establishments?
2. Why is this passing two separate int32s instead of a node id?  Or better yet a ScopedNodeId reference, since a node id on its own is not enough to say who established the CASE session.

Reply:
The CASE started and CASE Complete events will be emitted on every CASE establishment and as we are passing the ScopedNodeID, we will be able to track the node Id and fabric with which CASE Session is established.
The ScopedNodeId change has been made.

Reply:
Yes, but you won't be able to correlate the start and complete bits with each other.  Or start and failed.

Which keeps coming back to the question: what is the app supposed to _do_ with these notifications?

Reply:
The comment above clarifies the purpose of these notifications.